core/lib/spree/app_configuration.rb

```diff
@@ -209,6 +209,18 @@ class AppConfiguration < Preferences::Configuration
     #   @return [Integer] Orders to show per-page in the admin (default: +15+)
     preference :orders_per_page, :integer, default: 15
 
+    # @!attribute [rw] max_keys
+    #   @return [Integer] Maximum keys that can be allocated in customer and admin metadata column(default: +6+)
+    preference :max_keys, :integer, default: 6
```

Comment:
Can you explain why this preference could be helpful?

Reply:
It limits the maximum amount of key value pairs. 

As much xml expansion is not a problem here, we never the less given the low degree of auth needed to write practically anything create a security issue. 

Limiting it seemed reasonable. I will see to adapt the PR and Documentation accordingly as i believe it's important to have it there.

Reply:
I'm also not convinced this is a valid addition. Can you name one real world example where this might be needed?